control-plane/gateways/deployment_test.go

```diff
@@ -385,6 +406,375 @@ func Test_meshGatewayBuilder_Deployment(t *testing.T) {
+							Labels: meshv2beta1.GatewayClassAnnotationsLabelsConfig{
+								Set: map[string]string{
+									"app":      "consul",
+									"chart":    "consul-helm",
+									"heritage": "Helm",
+									"release":  "consul",
+								},
+							},
+						},
+						Deployment: meshv2beta1.GatewayClassDeploymentConfig{
+							Affinity: &corev1.Affinity{
+								PodAntiAffinity: &corev1.PodAntiAffinity{
+									PreferredDuringSchedulingIgnoredDuringExecution: []corev1.WeightedPodAffinityTerm{
+										{
+											Weight: 1,
+											PodAffinityTerm: corev1.PodAffinityTerm{
+												LabelSelector: &metav1.LabelSelector{
+													MatchLabels: map[string]string{
+														labelManagedBy: "consul-k8s",
+														"app":          "consul",
+														"chart":        "consul-helm",
+														"heritage":     "Helm",
+														"release":      "consul",
+													},
+												},
+												TopologyKey: "kubernetes.io/hostname",
+											},
+										},
+									},
+								},
+							},
+							Container: &meshv2beta1.GatewayClassContainerConfig{
+								HostPort:     8080,
+								PortModifier: 8000,
+							},
+							NodeSelector: map[string]string{"beta.kubernetes.io/arch": "amd64"},
+							Replicas: &meshv2beta1.GatewayClassReplicasConfig{
+								Default: pointer.Int32(1),
+								Min:     pointer.Int32(1),
+								Max:     pointer.Int32(8),
+							},
+							PriorityClassName: "priorityclassname",
+							TopologySpreadConstraints: []corev1.TopologySpreadConstraint{
+								{
+									MaxSkew:           1,
+									TopologyKey:       "key",
+									WhenUnsatisfiable: "DoNotSchedule",
+								},
+							},
+						},
+					},
+				},
+			},
+			want: &appsv1.Deployment{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: map[string]string{
+						labelManagedBy: "consul-k8s",
+						"app":          "consul",
+						"chart":        "consul-helm",
+						"heritage":     "Helm",
+						"release":      "consul",
+					},
+
+					Annotations: map[string]string{},
+				},
+				Spec: appsv1.DeploymentSpec{
+					Replicas: pointer.Int32(1),
+					Selector: &metav1.LabelSelector{
+						MatchLabels: map[string]string{
+							labelManagedBy: "consul-k8s",
+							"app":          "consul",
+							"chart":        "consul-helm",
+							"heritage":     "Helm",
+							"release":      "consul",
+						},
+					},
+					Template: corev1.PodTemplateSpec{
+						ObjectMeta: metav1.ObjectMeta{
+							Labels: map[string]string{
+								labelManagedBy: "consul-k8s",
+								"app":          "consul",
+								"chart":        "consul-helm",
+								"heritage":     "Helm",
+								"release":      "consul",
+							},
+
+							Annotations: map[string]string{
+								constants.AnnotationGatewayKind:                     meshGatewayAnnotationKind,
+								constants.AnnotationMeshInject:                      "false",
+								constants.AnnotationTransparentProxyOverwriteProbes: "false",
```

Comment:
If my pull request merges before yours, it will break this test because I added annotations. If yours merges first, I'll have to make the fix. Not a blocker, just a heads up.

Reply:
fixed em as your PR merged first